lib/canary/plugs.ex

```diff
@@ -181,6 +181,27 @@ defmodule Canary.Plugs do
   plug :load_resource, model: Post, id_name: "slug", id_field: "slug", only: [:show], persisted: true
   ```
   """
+  def authorize_controller(%{assigns: %{current_user: nil}} = conn, opts) do
```

Comment:
Why `%{current_user: nil}`? It looks like this makes the assumption that a `nil` user should always be unauthorized. However, this is not always true. For example, we might not assign a `current_user` for users that are not logged in, but we still want non-logged in users to be able to access certain controller actions.